rosbag2_performance/rosbag2_performance_benchmarking/src/result_utils.cpp

```diff
@@ -50,6 +50,31 @@ int get_message_count_from_metadata(const std::string & uri)
   return total_recorded_count;
 }
 
+size_t get_messages_recorded(
+  const rosbag2_storage::BagMetadata & metadata,
+  const PublisherGroupConfig & pub_config)
+{
+  size_t messages_recorded = 0;
+  for (unsigned int i = 0; i < pub_config.count; ++i) {
+    const std::string topic_without_node_name =
```

Comment:
@MichaelOrlov Is it guaranteed that this `topic_without_node_name` is unique in the `BagMetadata`? In other words, is it possible to have more than one `PublisherGroupConfig`, or otherwise create topics with the same name, but in different namespaces, etc?

Reply:
@kylemarcey In general the `topic_without_node_name` is **not** unique in the `BagMetadata`.
However, by common sense, it is supposed to be unique.  
The `PublisherGroupConfig` is defined in the producers yaml file as
https://github.com/ros2/rosbag2/blob/d788e518b3a93acb61226d993e3d3095913f16b1/rosbag2_performance/rosbag2_performance_benchmarking/config/producers/automotive.yaml#L4-L20 
and `topic_root` + publisher sequence number define the topic name.  
In theory, if someone by mistake will use the same `topic_root` for different publisher groups the calculation of the received messages will be incorrect and will consist of the total number of messages from the publisher group which has the same `topic_root`.
The `topic_root` is supposed to be unique across the publisher groups by design.

Reply:
@MichaelOrlov ok, reasonable to me to say that it relies on the user giving a sane configuration